src/test/scala/dev/kamu/cli/MetadataRepositorySpec.scala

```diff
@@ -53,4 +61,63 @@ class MetadataRepositorySpec extends FunSuite with Matchers with KamuTestBase {
       actual3 shouldEqual Seq(dsC.id, dsB.id, dsA.id)
     }
   }
+
+  test(raw"'kamu add' from HTTP") {
+    withEmptyRepo { kamu =>
+      val expected = DatasetFactory.newRootDataset()
+
+      // create a temporary directory with the dataset to host
+      val serverDir = getRandomDir()
+      fileSystem.mkdirs(serverDir)
+      val path: Path = new Path(serverDir, raw"test-dataset.yaml")
+      kamu.metadataRepository.saveDataset(expected, path)
+
+      // start up the server and host the directory
+      val serverPort = 80 // httpd:2.4 default port
+      val testServerName = s"kamu_test_http_server"
+      val testHttpServerArgs = DockerRunArgs(
+        image = "httpd:2.4",
+        exposePorts = List(serverPort),
+        volumeMap = Map(serverDir -> new Path("/usr/local/apache2/htdocs")), // httpd:2.4 default location
+        containerName = Some(testServerName),
+        detached = true
+      )
+      val testHttpServer = new DockerClient(fileSystem)
+      val testHttpServerProc = new DockerProcessBuilder(
+        "http",
+        testHttpServer,
+        testHttpServerArgs
+      ).run()
+      testHttpServerProc.join()
+      val hostPort =
+        testHttpServerProc.waitForHostPort(serverPort, Duration("500 ms"))
+
+      // pull the dataset from the server
+      val actual =
+        kamu.metadataRepository.loadDatasetFromURI(
+          new URI(s"http://localhost:${hostPort}/test-dataset.yaml")
+        )
+
+      // stop the server
+      testHttpServer.kill(testServerName)
```

Comment:
probably should be in a `finally` block not to "leak" the container

Reply:
got it